### PR TITLE
TDR-1596 - Display include a top level Folder checkbox on the upload …

### PR DESCRIPTION
**Pull request by vimleshtna**
…page and pass its value to the API

### CODE REVIEW

File: app/views/standard/upload.scala.html [hunk 1]
```diff
@@ -85,7 +85,18 @@ <h2 class="govuk-error-summary__title" id="error-summary-title">
                                 </div>
                             </div>
                             <p class="govuk-body">For information on what metadata will be captured during upload, visit the <a href="@routes.FaqController.faq()#metadata-captured" target="_blank" rel="noopener noreferrer" class="govuk-link">FAQ (opens in new tab)</a>.</p>
-
+                            <div class="govuk-form-group">
+                                <fieldset class="govuk-fieldset" aria-describedby="waste-hint">
+                                    <div class="govuk-checkboxes">
+                                        <div class="govuk-checkboxes__item">
+                                            <input class="govuk-checkboxes__input" id="includeTopLevelFolder" name="includeTopLevelFolder" type="checkbox" value="mines">
```
Comment (TomJKing) [inline app/views/standard/upload.scala.html:92]: Should the `@value` be set to `mines`?

File: app/views/standard/upload.scala.html [hunk 2]
```diff
@@ -85,7 +85,18 @@ <h2 class="govuk-error-summary__title" id="error-summary-title">
                                 </div>
                             </div>
                             <p class="govuk-body">For information on what metadata will be captured during upload, visit the <a href="@routes.FaqController.faq()#metadata-captured" target="_blank" rel="noopener noreferrer" class="govuk-link">FAQ (opens in new tab)</a>.</p>
-
+                            <div class="govuk-form-group">
+                                <fieldset class="govuk-fieldset" aria-describedby="waste-hint">
```
Comment (TomJKing) [inline app/views/standard/upload.scala.html:89]: Should the `@aria-describedby` be set to `waste-hint`?

File: build.sbt [hunk 3]
```diff
@@ -33,7 +33,7 @@ libraryDependencies ++= Seq(
   "com.softwaremill.sttp.client" %% "async-http-client-backend-future" % sttpVersion,
   "uk.gov.nationalarchives" %% "tdr-graphql-client" % "0.0.94",
   "uk.gov.nationalarchives" %% "tdr-auth-utils" % "0.0.120",
-  "uk.gov.nationalarchives" %% "tdr-generated-graphql" % "0.0.313",
+  "uk.gov.nationalarchives" %% "tdr-generated-graphql" % "0.0.314",
```
Comment (TomJKing) [inline build.sbt:36]: Is that the latest version of the graphql?

File: npm/src/upload/form/upload-form.ts [hunk 4]
```diff
@@ -63,6 +64,10 @@ export class UploadForm {
     return value
   }
 
+  includeTopLevelFolder: () => boolean = () => {
+    return this.formElement.includeTopLevelFolder?.checked
```
Comment (TomJKing) [inline npm/src/upload/form/upload-form.ts:68]: Will this return `false` if the box is not checked?

**Review by TomJKing — APPROVED** [review]
Looks good. Just some minor comments. Thanks

File: build.sbt (hunk 3 above)
Comment (vimleshtna) [reply build.sbt:36]: Updated with the latest version.

File: npm/src/upload/form/upload-form.ts (hunk 4 above)
Comment (vimleshtna) [reply npm/src/upload/form/upload-form.ts:68]: Yes, I tested it and it's working as expected.

**Pull request merged**